3_visualize/src/plot_gradient.R

```diff
@@ -0,0 +1,94 @@
+plot_gradient <- function(ensemble_data, site_info, date_start, days_shown, out_file){
+  
+  # filter data to a focal time period
+  date_start <- as.Date(date_start)
+  
+  threshold = 23.89 # C
+  
+  plot_df <- ensemble_data %>%
+    filter(model_name == 'DA') %>%
+    mutate(obs_max_temp_f = c_to_f(obs_max_temp_c),
+           pred_max_temp_f = c_to_f(max_temp)
+    ) %>%
+    # subset to time period of interest
+    filter(time %in% seq.Date(from = date_start, to = date_start+days_shown, by = "1 day")) %>%
+    # filter to 1 day out predictions only
+    filter(lead_time == 1) %>%
+    left_join(site_info)
```

Comment:
some of this formatting/filtering etc might be good to do outside for when we use it in other figs, but we can cross that bridge when we get there